Sources/Basic/JSON.swift

```diff
@@ -39,10 +39,17 @@ public enum JSON {
     case string(String)
 
     /// An array.
-    case array([JSON])
+    case array([JSONConvertible])
 
     /// A dictionary.
-    case dictionary([String: JSON])
+    case dictionary([String: JSONConvertible])
```

Comment:
I would rather not change this... arrays of protocol type are going to be more expensive than arrays of a known fixed type.